### PR TITLE
Fix travis

### PR DESCRIPTION
**Pull request by peter-edge**
Travis was broken with how it is set up with Docker. This fixes some environment variables, and gets coveralls working again, along with simplifying the Makefile so that `make dockerci` now does almost the same thing locally as it does remotely.

### CODE REVIEW

**Comment by glibsm** [comment]
`make ci` and `make travis` seem like the same thing now

**Comment by glibsm** [comment]
seems like bash syntax is still borked

File: Makefile [hunk 1]
```diff
@@ -137,6 +137,14 @@ dockerbuild:
 dockerrun: dockerbuild
 	docker run -e V -e COVERMODE $(DOCKER_IMAGE) make $(DOCKER_MAKE_CMD)
 
+.PHONY: travis
+travis: dockerbuild
+	if [ "$(DOCKER_GO_VERSION)" == "1.8" ]; then \
+		docker run -e V -e COVERMODE $(DOCKER_IMAGE) make coveralls; \
+	else; \
```
Comment (glibsm) [inline Makefile]: no need for this semi

**Comment by peter-edge** [comment]
Trying caching for now, it may slow it down, if so I'll remove it.

**Comment by alsamylkin** [comment]
Coveralls is working!:)

File: .dockerignore [hunk 2]
```diff
@@ -1 +1,5 @@
+.cache
+examples/dig/dig
```
Comment (madhuravi) [inline .dockerignore]: Why is this necessary? We want to make sure that examples still compile.

Comment (peter-edge) [inline .dockerignore]: This ignores them when building the docker container, ie it doesn't pass (usually darwin) binaries to the (linux) container. They are still built by `make ci` within the docker container.

Short: this has no effect on the build other than making the tarball sent to the docker daemon smaller. https://docs.docker.com/engine/reference/builder/#dockerignore-file
